torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -310,8 +310,87 @@ RegisterOperators reg({
     };                                                                 \
   }),
 
+// Operators that take two lists and return a bool
+#define DEFINE_LIST_COMPARISON_OP(aten_op, impl_name) \
+  Operator("aten::" #aten_op "(int[] a, int[] b) -> int", impl_name<Shared<IntList>>), \
+  Operator("aten::" #aten_op "(float[] a, float[] b) -> int", impl_name<Shared<DoubleList>>), \
+  Operator("aten::" #aten_op "(Tensor[] a, Tensor[] b) -> int", impl_name<Shared<TensorList>>),
+
+// Operators that take two lists and return a list of the same type
+#define DEFINE_LIST_FUNCTIONAL_OP(aten_op, impl_name) \
+  Operator("aten::" #aten_op "(int[] a, int[] b) -> int[]", impl_name<Shared<IntList>, int64_t>), \
+  Operator("aten::" #aten_op "(float[] a, float[] b) -> float[]", impl_name<Shared<DoubleList>, double>), \
+  Operator("aten::" #aten_op "(Tensor[] a, Tensor[] b) -> Tensor[]", impl_name<Shared<TensorList>, at::Tensor>),
+
+
+template <typename T>
+Operation listEq(Node* node) {
+  return [=](Stack& stack) {
+    T a;
+    T b;
+    pop(stack, a, b);
+    if (a->elements() == b->elements()) {
+      push(stack, 1);
+    } else {
+      push(stack, 0);
+    }
+    return 0;
+  };
+}
+
+// Specialization for at::Tensor, since it doesn't define operator==
+template <>
+Operation listEq<Shared<TensorList>>(Node* node) {
+  return [=](Stack& stack) {
+    Shared<TensorList> a;
+    Shared<TensorList> b;
+    pop(stack, a, b);
+    if (a->elements().size() != b->elements().size()) {
+      push(stack, 0);
+      return 0;
+    }
+
+    for (size_t i = 0; i < a->elements().size(); ++i) {
+      const auto& a_element = a->elements()[i];
+      const auto& b_element = b->elements()[i];
+      if (!a_element.equal(b_element)) {
```

Comment:
I'm pretty sure this is inconsistent with our Python semantics. IIRC it will actually be an error if the tensors have more than a single element

Reply:
Can you clarify? Right now I can do something like:
```
>> j = [torch.rand(i) for i in range(1,10)]
>> k = [torch.rand(i) for i in range(1,10)]
>> j == k
False
>> j == j
True
```

Reply:
Sure, but
```python
>>> import torch
>>> x = [torch.tensor([1, 1]) for _ in range(4)]                                                       
>>> y = [torch.tensor([1, 1]) for _ in range(4)]                                                       
>>> x == y
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: bool value of Tensor with more than one value is ambiguous
```

Reply:
Ah I see. Should I just make the TensorList specialization throw then?